app/imports/ui/pages/smart-contract/SignSmartContract.jsx

```diff
@@ -27,18 +29,55 @@ const bridgeTenantStance = new SimpleSchema2Bridge(contractSchemaTenantStance);
 
 const isTenant = (contract, username) => contract.tenantEmail === username;
 const isHomeowner = (contract, username) => contract.homeownerEmail === username;
+const contractPending = (contract) => contract.status === 'Pending';
 
 const missingSignature = (contract, username) => (isTenant(contract, username) && contract.tenantSignature === '') || (
   isHomeowner(contract, username) && contract.homeownerSignature === '');
 
 /** Renders a table containing all of the Stuff documents. Use <SmartContractItem> to render each row. */
 class SignSmartContract extends React.Component {
-  // On successful submit, insert the data.
+
+  // check if the contract has been signed by both parties
+  createContractObjs(profiles, data) {
+    const { homeownerEmail, tenantEmail } = data;
+    // create the objects to store data
+    const hOwner = createHomeowner(profiles, homeownerEmail);
+    const tNant = createTenant(profiles, tenantEmail);
+    console.log(hOwner, tNant);
+  }
+
+  signatureCheck(profiles, username, data) {
+    // update the contract and check if both people are signed, this should never run unless both are signed, or if I'm not a tenant nor homeowner
+    const { _id, homeownerEmail, tenantEmail } = data;
+    const newlySignedContract = SmartContracts.collection.findOne(_id);
```

Comment:
doesn't data contain the contract in question ? Not sure you need to fetch again.

Reply:
No it doesn't, when I was doing the initial test, it would only update after it would re render, I don't know why